### PR TITLE
Update nMinimumChainWork and defaultAssumeValid.

### PR DESCRIPTION
**Pull request by gmaxwell**

### CODE REVIEW

**Comment by petertodd** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
    Hash: SHA256

    utACK 3f78e460ba980d38149ee50ff5dfd75741aad9f2

    I believe 0000000000000000000000000000000000000000003f94d1ad391682fe038bf5 to a valid Bitcoin blockchain.

    -----BEGIN PGP SIGNATURE-----

    iQEcBAEBCAAGBQJYpgDFAAoJECSBQD2l8JH7xgEH/2n5BSrLNBkUrZthOeBYdXa+
    ntmjRzhZaMmlRgCTVnVJR0PN8e3FHEhdeWz+LxAwAlefOTehGqlqjxmaI1VQnqGq
    8a3LLvmZdQ5WD5fN4wGA4wDU1DjRPdG0k8O0W/p9BK/ABMgWiV0wyx1pDXvvn6V/
    I9r8CGNlxUd1VqT9x6T3rxSkEZIPZcYWkVSEfp930348Dz6W7wC9qFKw8+pZ95h0
    YVX/KUgFmq2+5ufuzBCW9irzwC1QrEcp1DO3Lg2myjYqqHJ2Z4wKN+zA2tNUpB+d
    8NxVbqBdyKdD0SbJMYYOV3byJwPBiiejZzFZgp5fu3obEOSS0JA8r+dMxLrxUl4=
    =Jkyt
    -----END PGP SIGNATURE-----

**Comment by jtimon** [comment]
utACK 3f78e46 for mainnet.
For testnet, ```getblock "0x00000000000128796ee387cf110ccb9d2f36cffaf7f73079c995377c65ac0dcc"``` tells me that the minimum chain work should be 0x00000000000000000000000000000000000000000000001dcfc5dbe39283e519 instead of 0x00000000000000000000000000000000000000000000001f057509eba81aed91

**Comment by MarcoFalke** [comment]
@jtimon The minimum chain work is not required to match the work of
the block that is assumed to be valid.

**Comment by jtimon** [comment]
Right, but any reason for not having them matching?

**Comment by MarcoFalke** [comment]
I assume the rationale is that testnet is super flaky and it is better
to be conservative in what block to assume valid, as it has to be part
of a valid chain to have any effect.

... In theory you could even "pre-date" the minimum chain work to the
minimum chain work you expect at the earliest expected time of
release.

**Comment by MarcoFalke** [comment]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Post merge utACK 3f78e460ba980d38149ee50ff5dfd75741aad9f
-----BEGIN PGP SIGNATURE-----

iQEcBAEBCgAGBQJYp58IAAoJEC1/I3LlD+E3lnMH/RYgl3ENy05TRDGxY8PMVkav
NVGIQCqgrsffHBQw62bCH66gSxpPnBLf5bOw8ZYhiHaiCMgw3ODx9DTc5GL/2v0V
m4F/Sr9PW4GQUKzVxTYZTGL+HkwYHplUtPr5yc9++pP8PkXHy/JzVp+4I6IwYAOr
qXEmGLrbNUfzvaUe9I+0SZYD/H+lFifg3b2FmX/i0hXJZMSqJ9sBQjCyHLGMBqU5
Do0eHAlp5kxtq0Dc4I+dzF22P5mOZ11xS6AtWUcD0o4MKQOq+1/gYh62jnWBuxHE
YbtVYaBg0gFZBFNLP8u4vi4LHbFPs6+DLuCQnO3ehXOcgOyOTMooZCFyxyF2Y08=
=eIBQ
-----END PGP SIGNATURE-----

File: src/chainparams.cpp [hunk 1]
```diff
@@ -97,10 +97,10 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_SEGWIT].nTimeout = 1510704000; // November 15th, 2017.
 
         // The best chain should have at least this much work.
-        consensus.nMinimumChainWork = uint256S("0x0000000000000000000000000000000000000000002cb971dd56d1c583c20f90");
+        consensus.nMinimumChainWork = uint256S("0x0000000000000000000000000000000000000000003f94d1ad391682fe038bf5");
 
         // By default assume that the signatures in ancestors of this block are valid.
-        consensus.defaultAssumeValid = uint256S("0x0000000000000000030abc968e1bd635736e880b946085c93152969b9a81a6e2"); //447235
+        consensus.defaultAssumeValid = uint256S("0x00000000000000000013176bf8d7dfeab4e1db31dc93bc311b436e82ab226b90"); //453354
```
Comment (sdaftuar) [inline src/chainparams.cpp]: ACK 0x00000000000000000013176bf8d7dfeab4e1db31dc93bc311b436e82ab226b90

File: src/chainparams.cpp [hunk 2]
```diff
@@ -201,10 +201,10 @@ class CTestNetParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_SEGWIT].nTimeout = 1493596800; // May 1st 2017
 
         // The best chain should have at least this much work.
-        consensus.nMinimumChainWork = uint256S("0x0000000000000000000000000000000000000000000000198b4def2baa9338d6");
+        consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000000000001f057509eba81aed91");
 
         // By default assume that the signatures in ancestors of this block are valid.
-        consensus.defaultAssumeValid = uint256S("0x000000000871ee6842d3648317ccc8a435eb8cc3c2429aee94faff9ba26b05a0"); //1043841
+        consensus.defaultAssumeValid = uint256S("0x00000000000128796ee387cf110ccb9d2f36cffaf7f73079c995377c65ac0dcc"); //1079274
```
Comment (sdaftuar) [inline src/chainparams.cpp]: ACK 0x00000000000128796ee387cf110ccb9d2f36cffaf7f73079c995377c65ac0dcc

**Review by sdaftuar — COMMENTED** [review]
post-merge ACK

**Pull request merged**